src/main/java/org/testng/TestNG.java

```diff
@@ -292,12 +295,18 @@ private void parseSuite(String suitePath) {
       for (XmlSuite s : allSuites) {
         s.setParallel(this.m_parallelMode);
         s.setThreadCount(this.m_threadCount);
-        // If test names were specified, only run these test names
-        if (m_testNames != null) {
-          m_suites.add(XmlSuiteUtils.cloneIfContainsTestsWithNamesMatchingAny(s, m_testNames));
-        } else {
+        if (m_testNames == null) {
           m_suites.add(s);
+          continue;
+        }
+        // If test names were specified, only run these test names
+        TestNamesMatcher testNamesMatcher = new TestNamesMatcher(s, m_testNames);
+        List<String> missMatchedTestname = testNamesMatcher.getMissMatchedTestNames();
+        if (!missMatchedTestname.isEmpty()) {
+          throw new TestNGException("The test(s) <" + Arrays.toString(missMatchedTestname.toArray())
```

Comment:
Same thing here as well. Please just do `throw new TestNGException("The test(s) <" + missMatchedTestname + "> cannot be found.");`

Reply:
Done.